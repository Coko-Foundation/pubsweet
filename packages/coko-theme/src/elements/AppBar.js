import { css } from 'styled-components'

/*
  To disable underline from Logo
*/

export default {
  LogoLink: css`
    &:hover:before {
      visibility: hidden;
    }
  `,
}
