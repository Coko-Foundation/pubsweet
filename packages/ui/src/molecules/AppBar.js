import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import Icon from '../atoms/Icon'

const Root = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-height: calc(var(--grid-unit) * 2);

  a {
    color: var(--color-primary);
    text-decoration: none;
  }
`

const Section = styled.div`
  display: flex;
`

const NavLinks = styled.div`
  align-items: center;
  display: flex;
  margin: 0 1rem;

  a {
    padding: 0 1rem;

    &:global(.active) {
      font-weight: bold;
    }
  }
`

const Logo = styled(Link)`
  display: flex;
  font-weight: bold;
  text-decoration: none;

  svg {
    height: calc(var(--grid-unit) * 2);
  }
`

const itemStyle = `
  align-items: center;
  display: inline-flex;
  padding: 0 1rem;

  svg {
    margin-right: 0.3rem;
  }
`

const Item = styled.span`
  ${itemStyle};
`
const ActionItem = Item.withComponent('a')
const LinkItem = styled(Link)`
  ${itemStyle};
`

const AppBar = ({
  brandLink = '/',
  brand,
  loginLink = '/login',
  onLogoutClick,
  navLinks,
  user,
  className,
}) => (
  <Root>
    <Section>
      {brand && <Logo to={brandLink}>{brand}</Logo>}

      {navLinks && <NavLinks>{navLinks}</NavLinks>}
    </Section>

    <Section>
      {user && (
        <Item>
          <Icon size={16}>user</Icon>
          {user.username}
          {user.admin ? ' (admin)' : ''}
        </Item>
      )}

      {user && (
        <ActionItem href="#" onClick={onLogoutClick}>
          <Icon size={16}>power</Icon>
          Logout
        </ActionItem>
      )}

      {!user && <LinkItem to={loginLink}>Login</LinkItem>}
    </Section>
  </Root>
)

export default AppBar
