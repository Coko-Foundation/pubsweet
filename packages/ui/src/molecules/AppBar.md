The app bar appears at the top of every page of the application.

It displays the name of the application (as a link to the home page), the
username of the current user, and a link to sign out.

```js
<AppBar brand="xpub" user={{ username: 'user', admin: true }} />
```

When the user is not signed in, only the login link is displayed.

```js
<AppBar brand="xpub" />
```

Can optionally pass navigation links or image for brand.

```js
<AppBar
  brand={
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="-220 -50 440 300"
      width="64"
      fillOpacity="0.5"
    >
      <circle cx="0" cy="50" fill="red" r="100" />
      <circle cx="70" cy="150" fill="blue" r="100" />
      <circle cx="-70" cy="150" fill="green" r="100" />
    </svg>
  }
  navLinkComponents={[
    <Link to="/home">Home</Link>,
    <Link to="/about">About</Link>,
  ]}
/>
```

Can use a custom component as right element.

```js
const RightComponent = ({ user, loginLink, onLogoutClick }) => (
  <div
    style={{
      padding: 10,
      border: '1px solid gray',
      display: 'flex',
      width: 200,
      justifyContent: 'space-between',
    }}
  >
    <span>{user ? user.username : 'admin'}</span>
    <button onClick={onLogoutClick}>logout</button>
  </div>
)

;<AppBar
  brand="xpub"
  user={{ username: 'userName', admin: true }}
  rightComponent={RightComponent}
  onLogoutClick={() => console.log('Logout clicked')}
/>
```
