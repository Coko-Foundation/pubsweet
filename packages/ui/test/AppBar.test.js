import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import AppBar from '../src/molecules/AppBar'
import TestThemeProvider from './setup/theme'

const baseProps = {
  brand: 'some brand',
  onLogoutClick: () => {},
}

const render = (props = {}) =>
  renderer
    .create(
      <MemoryRouter>
        <TestThemeProvider>
          <AppBar {...props} {...baseProps} />
        </TestThemeProvider>
      </MemoryRouter>,
    )
    .toJSON()

describe('AppBar', () => {
  test('Basic display', () => {
    const tree = render()
    expect(tree).toMatchSnapshot()
  })

  test('With admin user', () => {
    const tree = render({ user: { username: 'some user', admin: true } })
    expect(tree).toMatchSnapshot()
  })

  test('With nav links', () => {
    const tree = render({ navLinks: <div>Links!</div> })
    expect(tree).toMatchSnapshot()
  })

  test('With custom brand and login links', () => {
    const tree = render({ brandLink: '/home', loginLink: '/signin' })
    expect(tree).toMatchSnapshot()
  })
})
