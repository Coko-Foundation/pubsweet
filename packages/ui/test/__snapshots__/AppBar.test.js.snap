// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Basic display 1`] = `
.c4 {
  color: black;
  cursor: pointer;
  color: black;
  font: sans-serif;
  font-size: 18px;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c4:hover {
  background: none;
  color: black;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: calc(32px * 2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: calc(8px * 2) 1rem calc(8px * 2) 1rem;
}

.c3 > * {
  height: calc(32px * 2);
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: calc(32px * 1) 1rem calc(32px * 1) 0;
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      <a
        className="c3 c4"
        href="/"
        onClick={[Function]}
      >
        some brand
      </a>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c5"
    >
      <a
        className="c4"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </span>
  </div>
</nav>
`;

exports[`AppBar With admin user 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: calc(8px / 2);
}

.c6 svg {
  stroke: black;
  width: calc(2 * 8px);
  height: calc(2 * 8px);
}

.c7 {
  background: grey;
  border: 1px solid grey;
  border-radius: 8px;
  color: black;
  font-family: sans-serif;
  font-size: 18px;
  line-height: 32px;
  min-width: calc(32px * 4);
  padding: calc(32px / 2);
  background: none;
  border: none;
  min-width: 0;
  padding: 0;
  color: black;
  font: sans-serif;
  font-size: 18px;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c7:focus,
.c7:hover {
  background-color: #A1A1A1;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c7:active {
  background-color: #737373;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.c7[disabled]:focus,
.c7[disabled]:hover,
.c7[disabled]:active {
  background: grey;
}

.c7:hover {
  background: none;
  color: black;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4 {
  color: black;
  cursor: pointer;
  color: black;
  font: sans-serif;
  font-size: 18px;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c4:hover {
  background: none;
  color: black;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: calc(32px * 2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: calc(8px * 2) 1rem calc(8px * 2) 1rem;
}

.c3 > * {
  height: calc(32px * 2);
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: calc(32px * 1) 1rem calc(32px * 1) 0;
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      <a
        className="c3 c4"
        href="/"
        onClick={[Function]}
      >
        some brand
      </a>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c5"
    >
      <span
        className="c6"
        color="black"
        role="img"
        size={2}
      >
        <svg
          fill="none"
          height={undefined}
          stroke={undefined}
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={undefined}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="12"
            cy="7"
            r="4"
          />
        </svg>
      </span>
      some user
       (admin)
    </span>
    <span
      className="c5"
    >
      <span
        className="c6"
        color="black"
        role="img"
        size={2}
      >
        <svg
          fill="none"
          height={undefined}
          stroke={undefined}
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={undefined}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.36 6.64a9 9 0 1 1-12.73 0"
          />
          <line
            x1="12"
            x2="12"
            y1="2"
            y2="12"
          />
        </svg>
      </span>
      <button
        className="c7"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </span>
  </div>
</nav>
`;

exports[`AppBar With custom brand and login links 1`] = `
.c4 {
  color: black;
  cursor: pointer;
  color: black;
  font: sans-serif;
  font-size: 18px;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c4:hover {
  background: none;
  color: black;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: calc(32px * 2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: calc(8px * 2) 1rem calc(8px * 2) 1rem;
}

.c3 > * {
  height: calc(32px * 2);
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: calc(32px * 1) 1rem calc(32px * 1) 0;
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      <a
        className="c3 c4"
        href="/home"
        onClick={[Function]}
      >
        some brand
      </a>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c5"
    >
      <a
        className="c4"
        href="/signin"
        onClick={[Function]}
      >
        Login
      </a>
    </span>
  </div>
</nav>
`;

exports[`AppBar With custom right component 1`] = `
.c4 {
  color: black;
  cursor: pointer;
  color: black;
  font: sans-serif;
  font-size: 18px;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c4:hover {
  background: none;
  color: black;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: calc(32px * 2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: calc(8px * 2) 1rem calc(8px * 2) 1rem;
}

.c3 > * {
  height: calc(32px * 2);
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      <a
        className="c3 c4"
        href="/"
        onClick={[Function]}
      >
        some brand
      </a>
    </span>
  </div>
  <div>
    userName
  </div>
</nav>
`;

exports[`AppBar With nav links 1`] = `
.c4 {
  color: black;
  cursor: pointer;
  color: black;
  font: sans-serif;
  font-size: 18px;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.c4:hover {
  background: none;
  color: black;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: calc(32px * 2);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: calc(8px * 2) 1rem calc(8px * 2) 1rem;
}

.c3 > * {
  height: calc(32px * 2);
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: calc(32px * 1) 1rem calc(32px * 1) 0;
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      <a
        className="c3 c4"
        href="/"
        onClick={[Function]}
      >
        some brand
      </a>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c5"
    >
      <a
        className="c4"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </span>
  </div>
</nav>
`;
